### DISCUSSION TITLE
Error in installation

### DISCUSSION THREAD

**Discussion by jaslanli** [NOAA-EMC/WW3#1318] [Q&A] [answered]
Hello. I'm trying to install, compile and run test cases from regtests but facing with some problems with executables. 

I followed relevant steps described in [model/README.md](https://github.com/NOAA-EMC/WW3/blob/develop/model/README.md).
So I have relevant exectuable files in WW3/build directory. However, when I'm trying to run any regression test case (for example [ww3_tp1.1](https://github.com/NOAA-EMC/WW3/tree/develop/regtests/ww3_tp1.1), I get error in making makefile step: 

```
Checking all subroutines for modules (this may take a while) ...
ERROR: Error occured during WW3 ww3_grid build
```
Command I used is:

`./bin/run_test -g /mnt/c/Users/javid/Desktop/WW3/model/bin/ww3_grid -s PR1 -w work_PR1 /mnt/c/Users/javid/Desktop/WW3/model ww3_tp1.1`

I've tried several approaches to fix but can't fuigure out what is the main problem during installation stage as I'm quite new user of WW3. Could you please help me to have correct guidance for installation, compile and running a regression test case (ww3_tp1.1). 

Thanks!

**Comment by gonesurfing** [comment]
@jaslanli I was having all kinds of dependency issues on my system with different gcc/gfortran/cmake etc., so I made a docker image to isolate everything.  You can try cloning my repo at [https://github.com/gonesurfing/WW3](https://github.com/gonesurfing/WW3) and building the docker image . I was able to run the ww3_tp1.1 test doing this inside the container:

```
cd /WW3/regtests
../model/bin/w3_setup ../model -c gnu
./bin/run_test -s PR1 -w work_PR1 ../model ww3_tp1.1
```

It looks like you are using WSL, so it's entirely possible whatever default version of the gnu toolchain is installed is having issues. I haven't tried with the intel compiler yet.

**Comment by mickaelaccensi** [comment, marked as answer]
@jaslanli , you should use **run_cmake_test** instead of **run_test** for the regression tests.
For example, you can find some tutorials [here](https://gitlab.ifremer.fr/wave/Training/-/wikis/WW3-2023), like Tutorial WW3 compilation which include running a regression test.

Best,
